### PR TITLE
feat: persistent cli key and better get

### LINKED ISSUE #513: Stop using random keys
https://github.com/maidsafe/safe_network/blob/a30859dca3c5fbddb4e826c201d72bcfc4cc4e63/sn_cli/src/main.rs#L51

We use a different key at every command run. The direct implication I’m seeing is, now that registers require signatures, I get:
```
Error: Register error Access denied for user: Key(PublicKey(110f..2d1f)).
Error: Register error Access denied for user: Key(PublicKey(0fca..fb6c)).
...
```
Since we use a different key every time, if I create a register with: `safe register create myname`
I can’t write to it with a subsequent `safe register edit myname hello` because I have a new key, basically I loose access to it as soon as the command is over.

### PR DESCRIPTION
**Pull request by grumbach**
## Description

- fixes #513
- makes `safe register get [name]` print out the register value instead of just saying `Successfully retrieved Register`!

reviewpad:summary

### CODE REVIEW

File: sn_cli/src/subcommands/register.rs [hunk 1]
```diff
@@ -101,11 +101,22 @@ async fn get_registers(names: Vec<String>, client: &Client) -> Result<()> {
...
         match client.get_register(xorname, tag).await {
-            Ok(register) => println!(
-                "Successfully retrieved Register '{name}' from {}, {}!",
-                register.name(),
-                register.tag()
-            ),
+            Ok(register) => {
+                println!(
+                    "Successfully retrieved Register '{name}' from {}, {}!",
+                    register.name(),
```
Comment (maqi) [inline sn_cli/src/subcommands/register.rs]: using `register.address()` will give both.
actually, it shall be `address` to be used across to be consistent and avoid misleading

File: sn_cli/src/subcommands/register.rs [hunk 2]
```diff
@@ -101,11 +101,22 @@ async fn get_registers(names: Vec<String>, client: &Client) -> Result<()> {
...
+                    register.tag()
+                );
+                let entries = register.read();
```
Comment (maqi) [inline sn_cli/src/subcommands/register.rs:108]: better get `Register` with its `Debug` fmt refactored to giving those info ? if they are interested.
also, printly out all bytes might be not necessory, just first couple of hex will be enough and more readable?
like what we do for XorName Debug?

File: sn_client/src/register.rs [hunk 3]
```diff
@@ -58,8 +58,7 @@ impl ClientRegister {
 
     /// Create a new public Register (Anybody can write to it) and send it so the Network.
     pub async fn create_public_online(client: Client, name: XorName, tag: u64) -> Result<Self> {
```
Comment (maqi) [inline sn_client/src/register.rs:60]: better change `name: XorName, tag: u64` to `RegisterAddress` to avoid potential misusing

Comment (grumbach) [reply sn_client/src/register.rs:60]: `name: XorName, tag: u64` is used throughout the entire high level API, changing this here means changing our entire API. I don't have a strong opinion on this but we might want to think it through as it changes app devs interface. 
Probably out of scope for this PR.

File: sn_cli/src/subcommands/register.rs (hunk 2 above)
Comment (grumbach) [reply sn_cli/src/subcommands/register.rs:108]: Added a `address()` method to `ClientRegister`, be carefull the `register` here is a `ClientRegister` (includes a `Client`), not a `Register`

**Review by joshuef — APPROVED** [review]

**Pull request merged**